lib/commands/location.js

```diff
@@ -52,6 +52,10 @@ export default {
       return /** @type {Location} */ ({latitude, longitude, altitude: 0});
     }
 
+    if (this.opts.platformVersion && util.compareVersions(this.opts.platformVersion, '>=', '17.0')) {
+      this.log.warn(`Please use "mobile:setSimulatedLocation" to simulate the device location.`);
```

Comment:
What if we just change the logic and make it call this api instead of the one from ios-device for ios17+?
It won't be a breaking change

Reply:
When our appium-iOS-device gets an update for iOS 17+, probably it can do that via the ios-device, so still I haven't removed the existing way as no WDA dependency in this method so far

Reply:
it can but it would be safer to this change now since later Apple could remove usbmuxd completely and we'd have to drop ios-device support

Reply:
I'm ok to use WDA for ios 17+ only for now with logs (no strong opinion for this). I'll remove the code change from this docs pr right now to make it another pr.

ios-device may need to support `remoted` eventually, so _later Apple could remove usbmuxd completely_ yes for this possibility, or Apple will improve `devicectl` behavior then eventually it will be the ios-device replacement (maybe a couple of years later at least?) in device management perspective. (appium-remove-debugger usage may not tho)

Reply:
I'll do this change with a few more testing/checking in https://github.com/appium/appium-xcuitest-driver/pull/2062 tonight